include/com/centreon/exceptions/corruption.hh

```diff
@@ -16,27 +16,34 @@
 ** For more information : contact@centreon.com
 */
 
-#ifndef CCB_EXCEPTIONS_CORRUPTION_HH
-#define CCB_EXCEPTIONS_CORRUPTION_HH
+#ifndef CC_EXCEPTIONS_CORRUPTION_HH
+#define CC_EXCEPTIONS_CORRUPTION_HH
 
-#include "com/centreon/broker/exceptions/msg.hh"
-#include "com/centreon/broker/namespace.hh"
+#include <fmt/format.h>
+#include <exception>
+#include <string>
 
-CCB_BEGIN()
+#include "com/centreon/exceptions/msg_fmt.hh"
+#include "com/centreon/namespace.hh"
+
+CC_BEGIN()
 
 namespace exceptions {
 /**
  *  @class corruption corruption.hh
- * "com/centreon/broker/exceptions/corruption.hh"
+ * "com/centreon/exceptions/corruption.hh"
  *  @brief Shutdown exception class.
  *
  *  This exception is thrown when someone attemps to read from a
  *  stream that has been corruption.
  */
-class corruption : public msg {
+class corruption : public msg_fmt {
  public:
-  corruption() = default;
-  corruption(corruption const&) = default;
+  template <typename... Args>
+  explicit corruption(std::string const& str, const Args&... args)
+      : msg_fmt(fmt::format(str, args...)) {}
```

Comment:
As for config exception, we don't need fmt::format call here. It must be hidden within msg_fmt